relay-profiling/src/sample.rs

```diff
@@ -85,6 +85,34 @@ struct Profile {
 }
 
 impl Profile {
+    pub fn parse(&mut self, platform: &str, architecture: &str) -> Result<(), ProfileError> {
```

Comment:
The name of this function is maybe not really `parse` but more like `validate` or `normalize`.  

Also, please add docs.